package/MDAnalysis/converters/RDKitInferring.py

```diff
@@ -0,0 +1,699 @@
+
+            # switch the position of the charge and the double bond
+            matches = mol.GetSubstructMatches(pattern)
+            if matches:
+                # check if we haven't already transformed this triplet
+                for match in matches:
+                    # store order-independent atom indices
+                    g = set(match)
+                    # already transformed --> try the next one
+                    if g in backtrack:
+                        continue
+                    # add to backtracking and start the switch
+                    else:
+                        anion, a1, a2 = match
+                        backtrack.append(g)
+                        break
+                # already performed all changes
+                else:
+                    if backtrack_cycles == 1:
+                        # might be stuck because there were 2 "odd" end
+                        # patterns  misqualified as a single base one
+                        end_pattern = odd_end_pattern
+                    elif backtrack_cycles > 1:
+                        # likely stuck changing the same bonds over and over so
+                        # let's stop here
+                        mol.UpdatePropertyCache(strict=False)
+                        return
+                    match = matches[0]
+                    anion, a1, a2 = match
+                    backtrack = [set(match)]
+                    backtrack_cycles += 1
+
+                # switch charges
+                a = mol.GetAtomWithIdx(anion)
+                a.SetFormalCharge(a.GetFormalCharge() + 1)
+                a = mol.GetAtomWithIdx(a2)
+                a.SetFormalCharge(a.GetFormalCharge() - 1)
+                # switch bond orders
+                b = mol.GetBondBetweenAtoms(anion, a1)
+                b.SetBondType(RDBONDORDER[b.GetBondTypeAsDouble() + 1])
+                b = mol.GetBondBetweenAtoms(a1, a2)
+                b.SetBondType(RDBONDORDER[b.GetBondTypeAsDouble() - 1])
+                mol.UpdatePropertyCache(strict=False)
+                # update number of matches for the end pattern
+                n_matches = len(set([match[0] for match in matches]))
+                if n_matches == 1:
+                    end_pattern = odd_end_pattern
+                # start new iteration
+                continue
+
+            # no more changes to apply
+            mol.UpdatePropertyCache(strict=False)
+            return
+
+        # reached max_iter
+        warnings.warn(
+            "The standardization could not be completed within a "
+            "reasonable number of iterations"
+        )
+
+
+@dataclass(frozen=True)
+class TemplateInferer:
+    """Infer bond orders and charges by matching the molecule with a template
+    molecule containing bond orders and charges.
+
+    Attributes
+    ----------
+    template : rdkit.Chem.rdchem.Mol
+        Molecule containing the bond orders and charges.
+    adjust_hydrogens: bool, default = True
```

Comment:
Just checking that the default to `True` sounds reasonable to everyone? Otherwise if you have a charged mol you'll need to add explicit hydrogens on the template to have an exact match with your input mol which may be a bit of a pain.
This was originally in ProLIF to read PDBQT files as valid RDKit mols, happy to add it here (and should be fine license-wise)